fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LucenOnlineIndexingTest.java

```diff
@@ -937,5 +939,86 @@ void testRecordUpdateReducedMergeForcingAgileTimeQuota() {
+        int middle = 1 + rn.nextInt(numRecords - 2);
+        // write records
+        RecordMetaDataHook hook = metaDataBuilder -> {
+            metaDataBuilder.addIndex(MAP_DOC, srcIndex);
+            metaDataBuilder.addIndex(MAP_DOC, tgtIndex);
+            metaDataBuilder.removeIndex(TextIndexTestUtils.SIMPLE_DEFAULT_NAME);
+            TextIndexTestUtils.addRecordTypePrefix(metaDataBuilder);
+        };
+        try (final FDBRecordContext context = openContext()) {
+            recordStore = LuceneIndexTestUtils.openRecordStore(context, path, hook);
+            for (int i = 0; i < numRecords; i ++) {
+                long docId = docIds[i];
+                recordStore.saveRecord(createSimpleDocument(docId, randomText(rn), randomGroup(rn)));
+            }
+            context.commit();
+        }
+        // overwrite some records (to enforce merge), write others as new
+        try (final FDBRecordContext context = openContext()) {
+            recordStore = LuceneIndexTestUtils.openRecordStore(context, path, hook);
+            for (int i = 0; i < middle; i ++) {
+                long docId = docIds[i];
+                recordStore.saveRecord(createSimpleDocument(docId, randomText(rn), randomGroup(rn)));
+            }
+            for (int i = middle; i < numRecords; i ++) {
+                long docId = docIds[numRecords + i];
+                recordStore.saveRecord(createSimpleDocument(docId, randomText(rn), randomGroup(rn)));
+            }
+            context.commit();
+        }
+        try (final FDBRecordContext context = openContext()) {
+            recordStore = LuceneIndexTestUtils.openRecordStore(context, path, hook);
+            recordStore.markIndexDisabled(tgtIndex).join();
+            context.commit();
+        }
+        // build the index ..
+        try (final FDBRecordContext context = openContext()) {
+            recordStore = LuceneIndexTestUtils.openRecordStore(context, path, hook);
+            try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                    .setRecordStore(recordStore)
+                    .addTargetIndex(tgtIndex)
+                    .setLimit(transactionLimit)
+                    .setIndexingPolicy(OnlineIndexer.IndexingPolicy.newBuilder()
+                            .setReverseScanOrder(reverseScan)
+                            .setSourceIndex(srcIndex.getName())
+                            .build())
+                    .build()) {
+                assertTrue(recordStore.isIndexDisabled(tgtIndex));
+                indexBuilder.buildIndex(true);
+            }
+        }
+        // .. and assert readable mode
+        try (final FDBRecordContext context = openContext()) {
+            recordStore = LuceneIndexTestUtils.openRecordStore(context, path, hook);
+            assertTrue(recordStore.isIndexReadable(tgtIndex));
+        }
+        // assert number of segments
+        String[] allFiles = listFiles(tgtIndex);
+        assertTrue(allFiles.length < 12);
```

Comment:
It would probably be good to assert the validity of the index. I'm working on a utility in #2470 that would make that easier. Perhaps after both of these are merged we should update the index build tests to assert more about the validity of the resulting index.